app/Http/Controllers/Playlist/PlaylistsController.php

```diff
@@ -834,33 +835,37 @@ public function completeItem(Request $request, $item_id)
             $playlist_item = PlaylistItems::where('id', $item_id)->first();
 
             if (!$playlist_item) {
-                return $this->setStatusCode(SymfonyResponse::HTTP_NOT_FOUND)->replyWithError('Playlist Item Not Found');
+                return $this
+                    ->setStatusCode(SymfonyResponse::HTTP_NOT_FOUND)
+                    ->replyWithError('Playlist Item Not Found');
             }
 
-            $user_plan = UserPlan::join('plans', function ($join) use ($user) {
-                $join->on('user_plans.plan_id', '=', 'plans.id')->where('user_plans.user_id', $user->id);
-            })
-                ->join('plan_days', function ($join) use ($playlist_item) {
-                    $join
-                        ->on('plan_days.plan_id', '=', 'plans.id')
-                        ->where('plan_days.playlist_id', $playlist_item->playlist_id);
-                })
-                ->select('user_plans.*')
-                ->first();
+            $user_plan = UserPlan::getByPlaylistIdAndUserId($playlist_item->playlist_id, $user->id);
 
             if (!$user_plan) {
                 return $this->setStatusCode(SymfonyResponse::HTTP_NOT_FOUND)->replyWithError('User Plan Not Found');
             }
 
-            $complete = $complete && $complete !== 'false';
+            $result = null;
+
+            try {
+                $complete = $complete && $complete !== 'false';
 
-            if ($complete) {
-                $playlist_item->complete();
-            } else {
-                $playlist_item->unComplete();
+                if ($complete) {
+                    $playlist_item->complete();
+                } else {
+                    $playlist_item->unComplete();
+                }
+
+                $result = $complete ? 'completed' : 'not completed';
+            } catch(\Exception $e) {
```

Comment:
apply comment from above here also

Reply:
Hi @bradflood , I have also updated it to catch specifically the SQL constraint violations. See:

https://github.com/faithcomesbyhearing/dbp/pull/937/commits/50131cd2aaa5ca4a38819adf4bd5dbf8f8ecf668#diff-b2074cf830f20db0d05674376fd81b979af9a990752eb8e202431ca6729db2e4R864

Also, I have created a new class named MySQLErrorCode to configure the SQL constraint violations code.

[app/Exceptions/MySQLErrorCode.php](https://github.com/faithcomesbyhearing/dbp/pull/937/commits/50131cd2aaa5ca4a38819adf4bd5dbf8f8ecf668#diff-f4a1dcfb13aeada3d43aee6483f8f94507faff5ce6f27a10997a399e0192205dR3)